packages/components/pie-modal/src/defs.ts

```diff
@@ -59,6 +60,27 @@ export interface ModalProps extends RTLComponentProps {
     size: typeof sizes[number];
 
     /**
+     * The leading action configuration for the modal.
+     */
+    leadingAction: {
+        /**
+         * The text to display for the leading action button.
+         */
+        text: string;
+
+        /**
+         * The variant of the leading action button.
+         * Default: 'primary'
+         */
+        variant?: Variant;
+
+        /**
+         * The ARIA label for the leading action button.
+         */
+        ariaLabel?: string;
```

Comment:
I think it makes more sense for us to pass an aria object instead of a string primitive here as aria properties are not limited to a label only. for example, the button should accept the following aria attributes to provide means for an accessible button component (referenced from snacks button)
```
     aria-disabled={disabled}
      aria-label={aria.label}
      aria-labelledby={aria.labelledby}
      aria-describedby={aria.describedby}
      aria-expanded={aria.expanded}
      aria-controls={aria.controls}
      aria-live="polite"
      aria-busy={aria.loading ? 'true' : 'false'}
      aria-atomic="true"
```

```
  aria: {
    label: string,
    labelledby: string,
    describedby: string,
    loading: boolean,
    expanded: boolean,
    controls: string,
  },
```

this will require an update to the button itself as well, so feel free to address it in another PR if that makes more sense.